rmgpy/statmech/torsion.pyx

```diff
@@ -510,30 +510,45 @@ cdef class HinderedRotor(Torsion):
         should begin at zero and end at :math:`2 \pi`, with the minimum energy
         conformation having a potential of zero be placed at zero angle.
         """
-        cdef numpy.ndarray[numpy.float64_t,ndim=2] A
-        cdef numpy.ndarray[numpy.float64_t,ndim=1] b
-        cdef double phi
-        cdef int N, i, m
+        cdef numpy.ndarray[numpy.float64_t,ndim=2] A, fourier
+        cdef numpy.ndarray[numpy.float64_t,ndim=1] b, fit
+        cdef double phi, value, V0
+        cdef int N, i, m, numterms
+        numterms = 6
+        cdef bint NegativeBarrier
+        NegativeBarrier=True
+        # numterms is actually half the number of terms. It is called numterms 
+        # because it is the number of terms of either the cosine or sine fit
         
-        # Fit Fourier series potential
-        N = V.shape[0]
-        A = numpy.zeros((N+1,12), numpy.float64)
-        b = numpy.zeros(N+1, numpy.float64)
-        for i in range(N):
-            phi = angle[i]
-            for m in range(6):
-                A[i,m] = cos(m * phi)
-                A[i,6+m] = sin(m * phi)
-                b[i] = V[i]
-        # This row forces dV/dangle = 0 at angle = 0
-        for m in range(6):
-            A[N,m+6] = m
-        x, residues, rank, s = numpy.linalg.lstsq(A, b)
-        x *= 0.001
-        
-        self.fourier = ([x[1:6], x[7:12]], "kJ/mol")
+        while NegativeBarrier and numterms<18:
```

Comment:
Is there a reason you chose 18?  I'm not exactly sure what value should be used, but 18 seems a bit high to me.